cpp/src/arrow/python/numpy-internal.h

```diff
@@ -76,16 +76,12 @@ static inline std::string GetNumPyTypeName(int npy_type) {
     TYPE_CASE(INT16, "int16")
     TYPE_CASE(INT32, "int32")
     TYPE_CASE(INT64, "int64")
-#if (NPY_INT64 != NPY_LONGLONG)
```

Comment:
Note those inequalities wouldn't have the expected effect because of how macro expansion works (and I don't know how to fix that :-().